corehq/apps/smsforms/tests/test_app.py

```diff
@@ -155,3 +164,98 @@ def test_text_responses(self, tfsms_start_session_mock):
         session, responses = self._start_session(yield_responses=False)
 
         self.assertEqual(responses, ["human-error"])
+
+
+class TestFetchXml(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.session = SQLXFormsSession(session_id='abc123', domain='test-domain', user_id='user_id')
+        cls.formplayer_interface = FormplayerInterface(cls.session.session_id, cls.session.domain)
+
+    def setUp(self) -> None:
+        super().setUp()
+        patcher = patch.object(self.formplayer_interface, 'get_raw_instance')
+        self.mock_get_raw_instance = patcher.start()
+        self.addCleanup(patcher.stop)
+
+    def test_successful_response(self):
+        expected_result = '''
+        <data>
+            <question>answer</question>
+        </data>
+        '''.strip()
+
+        self.mock_get_raw_instance.return_value = {'status': 'success', 'output': expected_result}
+        result = _fetch_xml(self.formplayer_interface)
+        self.assertEqual(result, expected_result)
+
+    def test_returns_none_if_invalid_session_id(self):
+        self.mock_get_raw_instance.side_effect = InvalidSessionIdException
+        result = _fetch_xml(self.formplayer_interface)
+        self.assertIsNone(result)
+
+    def test_raises_touchforms_error_if_response_errors(self):
+        self.mock_get_raw_instance.return_value = {'status': 'error'}
+        with self.assertRaises(TouchformsError):
+            _ = _fetch_xml(self.formplayer_interface)
```

Comment:
nit: unnecessary assignment
```suggestion
            _fetch_xml(self.formplayer_interface)
```

Reply:
See b1a93ad6a157cf3cd6bf7cc5cc24f202a116dab8